### ISSUE TITLE
ofVideoPlayer::setPosition glitch on MacOS

### ISSUE THREAD

**Issue by kylemcdonald** [openframeworks/openFrameworks#6780]
The following code will work for the `fingers.mov` but when I try some other videos, sometimes it fails.

```c++
#pragma once

#include "ofMain.h"

class ofApp : public ofBaseApp{
	public:
    ofVideoPlayer player;
    void setup(){
        player.load("fingers.mov");
        player.play();
    }
    void update(){
        player.update();
    }
    void draw(){
        player.draw(0, 0, ofGetWidth(), ofGetHeight());
    }
    void mousePressed(int x, int y, int button) {
        player.setPosition(ofMap(x, 0, ofGetWidth(), 0, 1, true));
    }
};
```

One easy way to get a video that fails is to upscale fingers:

```
ffmpeg -i fingers.mov -vf scale=3840:-1 fingers-4k.mp4
```

Using that will cause the video playback to pause.

I have other videos with other codecs that will work (e.g. 4K Prores 422 LT) but if I add `player.update()` immediately after `player.setPosition()` then it will crash the app.

For all these videos I can easily seek in the QuickTime app with no latency, so this feels like a problem with the way OF uses the Mac SDK rather than a problem with my codec choice or videos.

**Comment by 2bbb** [comment]
@kylemcdonald

hi, kyle!

i tried to research this problem, but this code and fingers-4k.mp4 don't crash app on my envs.
what envs do you use?

FYI, my envs:
* both debug/release build
* oF 0.11.2
* Xcode 12.4
* Catalina 10.15.7
* MacBook Pro (15-inch, 2019)

**Comment by dimitre** [comment]
hello Kyle, I've upscaled fingers with ffmpeg and had intermittent issues decoding too. 
But the same issues appear on Quicktime/Preview also. In the same video positions.
Only VLC is able to play correctly this specific video after upscaling.

Other than that I think there is room for improvement in ofAVFoundationVideoPlayer too

**Comment by dimitre** [comment]
I've tested a lot AVFoundation player and in my usage the playback stops working if I seek a position before the actual position. and resumes after the playhead is equal to the position it was.
Today I found a fix which is commenting out this lines in ofAVFoundationVideoPlayer.m
```
	if (!bStream && (CMTimeCompare(time, videoSampleTime) < 0)) {
		// if jumping back in time
		[self createAssetReaderWithTimeRange:CMTimeRangeMake(time, duration)];
	}
```
And it works perfect here.
There appears to be some logic error in the variable videoSampleTime which prevents the videobuffer to be read.
